one-table.redshift.sql

```diff
@@ -0,0 +1,330 @@
+/*
+SQL Experiments for Typing and Normalizing AirbyteRecords in 1 table
+Run me on Postgres
+
+Schema:
+{
+  "id": "number",
+  "first_name": ["string", null],
+  "age": ["number", null],
+  "address": [null, {
+    "street": "string",
+    "zip": "string"
+  }],
+  "updated_at": timestamp
+}
+
+KNOWN LIMITATIONS
+* Only one error type shown per row, the first one sequentially
+* There's a full table scan used for de-duplication.  This can be made more efficient...
+* It would be better to show the actual error message from the DB, not custom "this column is bad" strings
+
+*/
+
+-- Set up the Experiment
+-- Assumption: We can build the table at the start of the sync based only on the schema we get from the source/configured catalog
+
+DROP TABLE IF EXISTS x.users;
+DROP TABLE IF EXISTS x.users_raw;
+
+CREATE TABLE x.users (
+    "_airbyte_raw_id" VARCHAR(36) NOT NULL -- Airbyte column, cannot be null
+    , "_airbyte_meta" super NOT NULL -- Airbyte column, cannot be null
+    , "_airbyte_extracted_at" timestamp NOT NULL -- Airbyte column, cannot be null
+    , "id" int8 -- PK cannot be null, but after raw insert and before typing, row will be null
+    , "first_name" text
+    , "age" int8
+    , "address" super
+    , "updated_at" timestamp
+);
+
+---------------------------------------
+--------- SAFE CAST METHODS -----------
+---------------------------------------
+
+-- DROP FUNCTION _airbyte_safe_cast_to_integer(input text)
+CREATE OR REPLACE FUNCTION _airbyte_safe_cast_to_integer(input text)
+RETURNS INTEGER
+STABLE AS $$
+	try:
+		v = int(input)
+		return v
+	except:
+		return None
+$$ LANGUAGE plpythonu;
+
+-- DROP FUNCTION _airbyte_safe_cast_to_boolean(input text)
+CREATE OR REPLACE FUNCTION _airbyte_safe_cast_to_boolean(input text)
+RETURNS BOOLEAN
+STABLE AS $$
+	try:
+		v = input.lower() == 'True' or input.lower() == 'true'
+		return v
+	except:
+		return None
+$$ LANGUAGE plpythonu;
```

Comment:
🟡 we need to test this.  I think the Airbyte Protocol sends booleans as literal `true` and `false`, but I'm worried about the SQL<-->python translation of the values, which is why I added the 'lower'